components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,179 @@
+
+/** Action sheet controllers must be created with actionSheetControllerWithTitle: or
+ with actionSheetControllerWithTitle:message:  */
+- (nonnull instancetype)initWithNibName:(NSString *)nibNameOrNil
+                                 bundle:(NSBundle *)nibBundleOrNil NS_UNAVAILABLE;
+
+/** Action sheet controllers must be created with actionSheetControllerwithTitle:
+   or with actionSheetControllerWithTitle:message:  */
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder NS_UNAVAILABLE;
+
+/**
+ Adds an action to the action sheet.
+
+ Actions are the possible reactions of the user to the presented alert. Actions are added as a
+ list item in the list of the action sheet.
+ Action buttons will be laid out from top to bottom depending on the the order they
+ were added.
+
+ @param action Will be added to the end of MDCActionSheetController.actions.
+ */
+- (void)addAction:(nonnull MDCActionSheetAction *)action;
+
+/**
+ The actions that the user can take in response to the action sheet.
+
+ The order of the actions in the array matches the order in which they were added to the action sheet.
+ */
+@property (nonatomic, nonnull, readonly) NSArray<MDCActionSheetAction *> *actions;
+
+/**
+ The title of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *title;
+
+/**
+ The message of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *message;
+
+/*
+ Indicates whether the button should automatically update its font when the device’s
+ UIContentSizeCategory is changed.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ If set to YES, this button will base its text font on MDCFontTextStyleButton.
+
+ Defaults value is NO.
+ */
+@property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
+    BOOL mdc_adjustsFontForContentSizeCategory;
+
+@end
+
+/**
+ MDCActionSheetActionHandler is a block that will be invoked when the action is selected.
+ */
+typedef void (^MDCActionSheetHandler)(MDCActionSheetAction *_Nonnull action);
+
+/**
+ An instance of MDCActionSheetAction is passed to MDCActionSheetController to add an
+ action to the action sheet.
+ */
+@interface MDCActionSheetAction : NSObject <NSCopying>
+
+
+/**
+ Returns an action sheet action with the populated given values.
+
+ @param title The title of the list item shown in the list
+ @param image The icon of the list item shown in the list
+ @param handler A block to execute when the user selects the action.
+ @return An initialized MDCActionSheetAction object.
+ */
++ (nonnull instancetype)actionWithTitle:(nonnull NSString *)title
+                                  image:(nonnull UIImage *)image
```

Comment:
The Apple version has a parameter for style, like "default", "cancel", or "destructive". Should we have our own versions of these?

Reply:
I think Apple uses this to render the actions differently, I believe all of our actions would render the same. But, if the team feels differently I'd love know that.

Reply:
I agree with @codeman7 here.  Since modal bottom sheets with action lists aren't expressly intended for confirmation of actions ([like a Dialog is](https://material.io/design/components/dialogs.html#usage)), I don't think we need to mark destructive/cancel actions. I would expect an action like "Delete" would result in a confirmation message or would support undo.